src/stp/plays/defensive/DefendShot.cpp

```diff
@@ -23,52 +23,118 @@ DefendShot::DefendShot() : Play() {
 
     roles = std::array<std::unique_ptr<Role>, stp::control_constants::MAX_ROBOT_COUNT>{
         std::make_unique<role::Keeper>(role::Keeper("keeper")),
-        std::make_unique<role::Defender>(role::Defender("defender_1")),
-        std::make_unique<role::Defender>(role::Defender("defender_2")),
-        std::make_unique<role::Harasser>(role::Harasser("harassing_defender")),
+        std::make_unique<role::Defender>(role::Defender("waller_1")),
+        std::make_unique<role::Defender>(role::Defender("waller_2")),
+        std::make_unique<role::Defender>(role::Defender("waller_3")),
+        std::make_unique<role::Harasser>(role::Harasser("defender_1")),
+        std::make_unique<role::Harasser>(role::Harasser("defender_2")),
+        std::make_unique<role::Harasser>(role::Harasser("harasser")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_1")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_2")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_3")),
+        std::make_unique<role::Defender>(role::Defender("midfielder_4")),
     };
 }
 
 uint8_t DefendShot::score(PlayEvaluator &playEvaluator) noexcept {
     auto world = playEvaluator.getWorld();
     auto field = world->getField().value();
     auto enemyRobot = world->getWorld()->getRobotClosestToBall(world::them);
+    auto goalVisibility = FieldComputations::getPercentageOfGoalVisibleFromPoint(field, true, enemyRobot->get()->getPos(), world->getWorld().value(), enemyRobot->get()->getId(), false);
     auto position = distanceFromPointToLine(field.getBottomLeftCorner(), field.getTopLeftCorner(), enemyRobot->get()->getPos());
-    return 255 * (field.getFieldLength() - position) / field.getFieldLength();
+    return 255 * (field.getFieldLength() - position) / field.getFieldLength() * goalVisibility/100;
```

Comment:
Perhaps we could use the GoalShotScore for this? Something like PositionScoring::scorePosition(enemyRobot->get->getPos() , gen::GoalShot, field, world).score;

This is also what we do in our Attack play, so it might make sense to employ a similar strategy here